cmake/environmentchecks.cmake

```diff
@@ -55,6 +55,26 @@ mark_as_advanced(
   VALGRIND_EXE
 )
 
+# Use clcache/ccache gets used, if available.
+if(WIN32)
+  find_program(CLCACHE clcache)
+  mark_as_advanced(CLCACHE)
+  if(CLCACHE)
```

Comment:
I think we should add a condition here that allows us to not use ccache/clcache if desired.

Reply:
What would be the rationale? I mean, unlike Clang-Tidy, for instance, it's not going to be slow things down (hence we have made the use of Clang-Tiydy optional). On the contrary, it's going to speed them up.

Reply:
It will wrap the compile command I can see this at times being very undesirable when developing.

Reply:
I really can't see why, but ok... done in commit db1b5cc.

Reply:
anything that abstracts the compiler must be optional, right? Otherwise you never know if its the wrapper or the compiler. The default can be configured based on having it available or not, but the user must have the option to turn it off without having to uninstall clcache...